### ISSUE TITLE
Support .cmt files too

### ISSUE THREAD

**Issue by leostera** [ostera/bsdoc#1]
While migrating https://github.com/ostera/reactor to `bsdoc` I realized that I was using the script there to build docs for the example modules but relying on the `.cmt` instead of the `.cmti` files.

I'd like this to be configurable, or use both by default (prefer `.cmti` over `.cmt` for same file name).

I'll let this rest for a bit and see if the problem arises again in other places.

**Comment by yawaramin** [comment]
I just ran into this in https://reasonml.chat/t/seeking-guidance-on-odoc-bsdoc-module-linking-on-bucklescript-project/2389 . I know you are not prioritizing bsdoc right now, I will try to look into fixing it. (As we know e.g. https://github.com/b0-system/odig/issues/2 , it is important to support `.cmt` in case the interface `.cmti` is missing.)

**Comment by yawaramin** [comment]
Hmm, I'm confused, because it seems that https://github.com/reuniverse/bsdoc/commit/2840121528d5b41752ef88b8aff0caacf78528e6 should have fixed this. Will try to make and run locally to see what's going on.

**Comment by yawaramin** [comment]
I think I understand what's going on now. In https://github.com/reuniverse/bsdoc/blob/0ab00a4116e12bb83d571de24d1e166dd164c3fc/src/odoc.re#L8 , every `.cmt`/`.cmti` file is compiled to odoc with the same single include directory (effectively, `-I lib/bs/src`). But, in projects organized in a nested directory structure, the nested directories need to be included too. Otherwise the `.odoc` files in those nested directories are not linked in to the final output, and their modules are not hyperlinked either.

So in short, this issue is resolved by the above commit, but I am seeing effectively a new issue, which is that `.odoc` files in nested directories are not linked.

**Comment by yawaramin** [comment]
Fixed as of 6.0.4-alpha
